### PR TITLE
FT API with Headlines

### LINKED ISSUE #5: Weeks 7-8
- [x] wrap up DB work - `res.render` a different message if the user already exists. Watch out for order of execution with the callback
- [x] building a new page (template) to display a list of headlines
- [x] building the endpoint on the server that's going to `res.render` those headlines
- [x] getting set up with the dev api
- [x] calling the headlines endpoint from the server

Additional tasks you could do:
- [x] styling that page
- [ ] experiment with pagination. Have a look at how FT do it (https://www.ft.com/world?format=&page=2) and what it would entail, front end and back end wise to accomplish it.

### PR DESCRIPTION
**Pull request by j-mes**
Closes #5.

- Add new endpoint at `/headlines`
- Add a POST request and uses isomorphic-fetch to get the latest news with 'Trump', and is limited to 10 headline items for the time being which can be found inside the controllers folder - `get-ft-headlines.js`.
- Add CSS styling with static files set to public folder.
- Move `root/server/views/` to `root/views/` and this fixes the issue with head not being imported correctly.
-  Add `headlines.html` which uses handlebars' each helper to populate the headlines from the POST request into an unordered list with list items.

This addresses all the points in the issue, apart from pagination which will be addressed in a separate PR.

### CODE REVIEW

**Comment by j-mes** [comment]
@lc512k I now have come across a blocker 😞 .

When I added in the fetch library and ran an API call, and it would connect to it fine however would return a bad request i.e. like this below:

```🔔 => Body {  url: 'http://api.ft.com/content/search/v1',
  status: 400,
  statusText: 'Bad Request',
  headers:
   Headers {     _headers:
      { age: [Array],
        'cache-control': [Array],
        'content-type': [Array],
        expires: [Array],
        pragma: [Array],
        server: [Array],
        via: [Array],
        'x-last-built': [Array],
        'x-ratelimit-limit': [Array],
        'x-ratelimit-remaining': [Array],
        'x-ratelimit-reset': [Array],
        'content-length': [Array],
        'accept-ranges': [Array],
        date: [Array],
        connection: [Array],
        'x-served-by': [Array],
        'x-cache': [Array],
        'x-cache-hits': [Array],
        'x-timer': [Array],
        vary: [Array] } },
  ok: false,
  body:
   PassThrough {
     _readableState:
      ReadableState {
        objectMode: false,
        highWaterMark: 16384,
        buffer: BufferList { length: 1 },
        length: 60,
        pipes: null,
        pipesCount: 0,
        flowing: null,
        ended: true,
        endEmitted: false,
        reading: false,
        sync: false,
        needReadable: false,
        emittedReadable: true,
        readableListening: false,
        resumeScheduled: false,
        emitClose: true,
        destroyed: false,
        defaultEncoding: 'utf8',
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: true,
     _events: { prefinish: [Function: prefinish] },
     _eventsCount: 1,
     _maxListeners: undefined,
     _writableState:
      WritableState {
        objectMode: false,
        highWaterMark: 16384,
        finalCalled: false,
        needDrain: false,
        ending: true,
        ended: true,
        finished: true,
        destroyed: false,
        decodeStrings: true,
        defaultEncoding: 'utf8',
        length: 0,
        writing: false,
        corked: 0,
        sync: false,
        bufferProcessing: false,
        onwrite: [Function: bound onwrite],
        writecb: null,
        writelen: 0,
        bufferedRequest: null,
        lastBufferedRequest: null,
        pendingcb: 0,
        prefinished: true,
        errorEmitted: false,
        emitClose: true,
        bufferedRequestCount: 0,
        corkedRequestsFree: [Object] },
     writable: false,
     allowHalfOpen: true,
     _transformState:
      { afterTransform: [Function: bound afterTransform],
        needTransform: false,
        transforming: false,
        writecb: null,
        writechunk: null,
        writeencoding: 'buffer' } },
  bodyUsed: false,
  size: 0,
  timeout: 0,
  _raw: [],
  _abort: false }
```
Pardon for the really long error message :D Would this be due to CORS or is that something that does not need to be set since it's being rendered server-side?

If I switch the method to 'GET', it'll come back with '405 - Method not allowed' which I assume is absolutely correct as from what I've seen with Postman, it uses POST with the collection of queries with the correct environment values (identical in this node app by the way). I'm trying to get past this 400 so I can then start getting the values and then drop them in the handlebars foreach block.

**Update 1:** Tried 'application/json' in the 'Content-Type', returned a 400 still. Hm, I think I'm missing something here.

**Update 2:** Removed the body from the options and ran it again. It returned a 400, so I assume it is either my authentication that's not correct, or maybe CORS?

File: server/controllers/get-ft-headlines.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const main = (req, res) => {
+
+	// process.env.FT_API = key
+	const fetch = require('isomorphic-fetch');
```
Comment (lc512k) [inline server/controllers/get-ft-headlines.js]: Best to do this at the top of the file, so it's not imported on every call

File: server/controllers/get-ft-headlines.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	let options = {
+		method: 'POST',
+		body: {
```
Comment (lc512k) [inline server/controllers/get-ft-headlines.js]: The body seems correct, but it needs to be a string 

`body: JSON.stringify({...})`

See https://github.com/github/fetch#post-json

Almost there!

**Comment by j-mes** [comment]
I've added in a loop that I'm happy with for the time being which is limited to 10 headline items. I'll be investigating a bit into the pagination tonight 👍 and adding some rudimentary styling so it gets rid of the ugly dots from the `ul/li` :)

**Comment by j-mes** [comment]
Updated the initial PR comment with context 👍

File: public/style.css [hunk 3]
```diff
@@ -0,0 +1,82 @@
+:root {
+	--primary-colour: #CCC;
+	--secondary-colour: #555;
+	--link-colour: #000;
+	--link-colour-alt: #FFDB3A;
```
Comment (lc512k) [inline public/style.css:5]: nice use of css vars :)

File: server/controllers/get-ft-headlines.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+			'queryString': 'Trump',
+			'resultContext': {
+				'maxResults': 10,
+				'aspects': ['title', 'lifecycle', 'location', 'summary', 'editorial']
+			}
+		})
+	};
+
+	fetch('http://api.ft.com/content/search/v1', options)
+	.then((res) => {
```
Comment (lc512k) [inline server/controllers/get-ft-headlines.js:21]: in a separate PR lets try async/await here

File: views/headlines.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<h1>{{pageTitle}}</h1>
+<ul class="headlines">
+	{{#each results}}
```
Comment (lc512k) [inline views/headlines.html:3]: nailed it. #each comes in very handy for this. We use it often

**Review by lc512k — APPROVED** [review]
Well done!

File: views/headlines.html (hunk 5 above)
Comment (j-mes) [reply views/headlines.html:3]: I found by using the `{{#each <object }}` to be really pleasant to work with.

File: public/style.css (hunk 3 above)
Comment (j-mes) [reply public/style.css:5]: I'm starting to appreciate CSS variables a little bit more now, especially for base stuff like colours and typography!

**Pull request merged**